### ISSUE TITLE
Incorrect `unused_variable` inspection with named parameters in format string

### ISSUE THREAD

**Issue by gshipilov** [intellij-rust/intellij-rust#9060]
## Environment

* **IntelliJ Rust plugin version:** 0.4.174.4743-221
* **Rust toolchain version:** 1.62.0 (a8314ef7d 2022-06-27) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.1.3 (CL-221.5921.27)
* **Operating system:** macOS 12.4
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
The rust plugin incorrectly flags variables that are only used inside a format string's named parameter as unused. The plugin _does_ correctly highlight the variable inside the `{}`, but other actions like renaming the variable don't work.

## Steps to reproduce
```rust
fn my_fn() {
    let foo = "hello, world!"; // "Variable 'foo' is never used."
    println!("{foo}"); // Prints "hello, world!" as expected.

    let bar_opt = Some("so long");
    if let Some(bar) = bar_opt { // "Variable 'bar' is never used."
        println!("{bar}"); // Prints "so long" as expected.
    }
}
```

**Comment by neonaot** [comment]
Duplicate of https://github.com/intellij-rust/intellij-rust/issues/8417
